lib/abci/handlers/query/verifyChainlockQueryHandlerFactory.js

```diff
@@ -0,0 +1,41 @@
+const { ChainLock } = require('@dashevo/dashcore-lib');
+const SimplifiedMNListStore = require('@dashevo/dashcore-lib/lib/deterministicmnlist/SimplifiedMNListStore');
+
+const {
+  abci: {
+    ResponseQuery,
+  },
+} = require('abci/types');
+
+/**
+ *
+ * @param {SimplifiedMNListStore} smlStore
+ * @return {verifyChainlockQueryHandler}
+ */
+function verifyChainlockQueryHandlerFactory(smlStore) {
+  /**
+   * @typedef identityQueryHandler
+   * @param {Object} params
+   * @param {Object} data
+   * @param {Buffer} data.chainLock
+   * @return {Promise<ResponseQuery>}
+   */
+  async function verifyChainlockQueryHandler(params, { chainLock }) {
+
+    const chainlock = new ChainLock(chainLock);
+
+    if (!chainlock.verify(smlStore)) {
+      return new ResponseQuery({
```

Comment:
You don't need specifically to return an errored response. You can just throw `InvalidArgumentAbciError. Take a look at other query handlers.

Reply:
done